src/TrackerCouncil.Smz3.Tracking/VoiceCommands/ChatIntegrationModule.cs

```diff
@@ -509,14 +510,33 @@ private void ChatClient_Connected(object? sender, EventArgs e)
         TrackerBase.Say(x => x.Chat.WhenConnected);
     }
 
-    private void ChatClient_Disconnected(object? sender, EventArgs e)
+    private async void ChatClient_Disconnected(object? sender, EventArgs e)
     {
-        TrackerBase.Say(x => x.Chat.WhenDisconnected);
+        try
+        {
+            waitingOnReconnect = true;
+            await Task.Delay(TimeSpan.FromSeconds(15));
+            if (waitingOnReconnect)
+            {
+                TrackerBase.Say(x => x.Chat.WhenDisconnected);
+            }
+        }
+        catch (Exception ex)
+        {
+            Logger.LogError(ex, "Unable to wait for reconnection");
+        }
     }
 
     private void ChatClient_Reconnected(object? sender, EventArgs e)
     {
-        TrackerBase.Say(x => x.Chat.WhenReconnected);
+        if (waitingOnReconnect)
+        {
+            waitingOnReconnect = false;
+        }
+        else
+        {
+            TrackerBase.Say(x => x.Chat.WhenReconnected);
```

Comment:
I think we won't get here unless we have two reconnect events in a row? Do we need a separate state flag for when the disconnect line has been said?

Reply:
Ah, you're right. I think when it's says the "WhenDisconnected" line, it should mark it as no longer waiting on a reconnect.